backend/src/main/scala/bloop/CompilerCache.scala

```diff
@@ -15,13 +15,14 @@ class CompilerCache(componentProvider: ComponentProvider,
                     logger: Logger,
                     userResolvers: List[Resolver]) {
 
-  private val cache = new ConcurrentHashMap[ScalaInstance, Compilers]()
+  private type CompilerID = (ScalaInstance, ClasspathOptions)
+  private val cache = new ConcurrentHashMap[CompilerID, Compilers]()
```

Comment:
Why is this necessary? It's not clear to me why classpath options is part of the compiler key, this is not nice performance wise because every time the classpath options change we need to instantiate a new compiler. Can we avoid this change?